syntax = "proto3";

service AgentService {
  rpc ProcessChatMessage(ChatMessage) returns (stream ChatMessageProgress);
}

message ChatMessage {
  string message = 1;
  string sender = 2;
}

message ChatMessageProgress {
  string status = 1;
  string filename = 2;
}
